docs/migrate-to-jina3/migrate.md

```diff
@@ -0,0 +1,185 @@
+  Additionally, you can access a specific key in `tags` by calling `docs[:'tags__key']`.
+
+**Array Traversal**: For traversing `DocumentArray`s via a `traversal_path`, docarray introduces a simplified notation:
+
+- Traversal paths of the form `[path1, path2]` (e.g. `['r', 'cm']`) are replaced by a single string of the form
+`'path1,path2'` (e.g. `'r,cm'`)
+- `docs.traverse_flat(path)` is replaced by `docs['@path']` (e.g. `docs['@r,cm']`)
+- `docs.flatten()` is replaced by `docs[...]`
+
+````{tab} Jina 2
+
+```python
+from Jina import Document, DocumentArray
+
+docs = nested_docs()
+
+print(docs.traverse_flat('r,c').texts)
+>>> ['root1', 'rooot2', 'chunk11', 'chunk12', 'chunk21', 'chunk22']
+
+print(docs.flatten().texts)
+>>> ['chunk11', 'chunk12', 'root1', 'chunk21', 'chunk22', 'root2']
+
+```
+
+````
+
+````{tab} Jina 3 
+
+```python
+from Jina import Document, DocumentArray
+
+docs = nested_docs()
+
+print(docs['@r,c'].texts)
+>>> ['root1', 'rooot2', 'chunk11', 'chunk12', 'chunk21', 'chunk22']
+
+print(docs[...].texts)
+>>> ['chunk11', 'chunk12', 'root1', 'chunk21', 'chunk22', 'root2']
+
+```
+
+````
+
+**Batching**: Batching operations are delegated to the docarray package and Python builtins:
+
+- `docs.batch()` does not accept the arguments `traversal_paths=` and `require_attr=` anymore.
+The example below shows how to achieve complex behavior that previously relied on these arguments, in a more pythonic
+and Jina 3 compatible way:
+
+````{tab} Jina 2
+
+```python
+docs.batch(traversal_paths=paths, batch_size=bs, require_attr='attr')
+```
+
+````
+
+````{tab} Jina 3 
+
+```python
+DocumentArray(filter(lambda x : bool(x.attr), docs['@paths'])).batch(batch_size=bs)
+```
+
+````
+
+## Method behavior changes
+
+- **`.post()` method**: `flow.post()` now returns a flattened `DocumentArray` instead of a list of `Response`s, if `return_results=True` is
+set. This makes it easier to immediately use the returned results. The behavior of `client.post()` remains unchanged
+compared to Jina 2, exposing entire `Response`s to the user. By setting or unsetting the `results_as_docarray=` flag,
+the user can override these default behaviors.
+- **Accessing non-existent values**: In Jina 2, bulk accessing attributes in a `DocumentArray` returns a list of empty values, when the `Document`s
+inside the `DocumentArray` do not have a value for that attribute. In Jina 3, this returns `None`. This change becomes
+important when migrating code that checks for the presence of a certain attribute.
+
+````{tab} Jina 2
+
+```python
+from Jina import Document, DocumentArray
+
+d = Document()
+print(d.text)
+>>> ''
+
+docs = DocumentArray([d, d])
+print(docs.texts)
+>>> ['', '']
+```
+
+````
+
+````{tab} Jina 3 
+
+```python
+from Jina import Document, DocumentArray
+
+d = Document()
+print(d.text)
+>>> ''
+
+docs = DocumentArray([d, d])
+print(docs.texts)
+>>> None
+```
+
+````
+
+
+## Pythonic serialization
```

Comment:
Try to group the changes between `Document`, `DocumentArray` based changes and the `Flow` ones.